### PR TITLE
Parameterize build, remove redundant step

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Remove redundant build step.
Simplify/Standardize config with parameters.
Currently push new docker.io/libra/build_environment:circleci-latest whenever a file that would modify the image is committed and lands in master
Will need a second commit to update the build to the new tag once this lands.

### Have you read the [Contributing Guidelines on pull requests]
Yes

## Test Plan

CI, Folding the build artifacts like origami.

## Related PRs

None

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1037,60 +893,43 @@ workflows:
...
               ignore:
-                - auto
                 - /^test-[\d|.]+$/
                 - /^release-[\d|.]+$/
                 - master
-      - dev_setup-test-centos:
-          requires:
-            - dev-setup-changes
+      - docker-ci-build:
+          base-image: centos
```
Comment (sausagee) [inline .circleci/config.yml:912]: Why are we building these variants?

File: .circleci/config.yml [hunk 2]
```diff
@@ -653,165 +660,20 @@ jobs:
...
-              echo no relevant files have changed - exit
-              circleci step halt
-            fi
-            set -x
-            #building with content trust tends to break in circleci
-            export DOCKER_CONTENT_TRUST=0
-            docker build -f docker/ci/Dockerfile -t libra/build_environment:latest .
             #signs the pushed image
-            docker push --disable-content-trust=false libra/build_environment:latest
+            docker push --disable-content-trust=false libra/build_environment:<< parameters.base-image >>-latest
```
Comment (sausagee) [inline .circleci/config.yml:676]: Need to think about DockerHub image retention policy soon :)

File: .circleci/config.yml (hunk 1 above)
Comment (rexhoffman) [reply .circleci/config.yml:912]: to test the dev_setup.sh that claims to support centos.  now we know.

File: .circleci/config.yml (hunk 2 above)
Comment (rexhoffman) [reply .circleci/config.yml:676]: I'm aware.   This only builds on changes to the relevant files.

Comment (rexhoffman) [reply .circleci/config.yml:676]: I'm not done with the nightly run yet that will be triggered on master to handle the cleanups.   It's a bit of a pia, and got side tracked with lcov stuff today.

**Pull request closed without merging**